packages/shared/src/api/blizzardApi.ts

```diff
@@ -0,0 +1,74 @@
+import LRUCache from 'lru-cache';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+// Tokens are region specific!
+const tokens: Record<string, string> = {
+  eu: 'invalid',
+  us: 'invalid',
+  apac: 'invalid',
+  cn: 'invalid',
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const apiCache = new LRUCache<string, any>({
+  maxAge: 1000 * 60 * 60, // 60m
+  maxSize: 100 * 1024 * 1024,
+  sizeCalculation: (e) => JSON.stringify(e).length,
+});
+
+const clientId = process.env.BLIZZARD_CLIENT_ID;
+const clientSecret = process.env.BLIZZARD_CLIENT_SECRET;
+
+function createBlizzardFetch(route: string[], namespace: string, locale: string) {
+  const froute = `https://${route[0]}.api.blizzard.com/${route
+    .slice(1)
+    .join('/')}?namespace=${namespace}&locale=${locale}`;
+  return (token: string) =>
+    fetch(encodeURI(froute + `&access_token=${token}`), {
+      headers: {
+        Accept: 'application/json',
+      },
+    });
+}
+
+interface AuthResult {
+  access_token: string;
+  expires_in: string;
+  sub: string;
+  token_type: string;
+}
+
+async function refreshToken(region: string): Promise<string> {
+  const apiResult = await fetch(`https://${region}.battle.net/oauth/token?grant_type=client_credentials`, {
```

Comment:
i vaguely remember this should be region-less now. see https://oauth.battle.net/.well-known/openid-configuration. have you tried?

Reply:
gonna take a look

Reply:
looks like only the oauth endpoint went regionless :\